src/connectors/numeric-menu/connectNumericMenu.ts

```diff
@@ -126,25 +133,64 @@ const connectNumericMenu: NumericMenuConnector = function connectNumericMenu(
       }));
 
     const connectorState: ConnectorState = {};
+    let sendEvent;
 
     return {
       $$type: 'ais.numericMenu',
 
       init({ helper, createURL, instantSearchInstance }) {
+        sendEvent = (...args) => {
```

Comment:
this function should probably be in utils as well like createSendEventForFacet

Reply:
- [connectNumericMenu](https://github.com/algolia/instantsearch.js/blob/feat/send-events-2/src/connectors/numeric-menu/connectNumericMenu.ts#L142-L172)
- [connectToggleRefinement](https://github.com/algolia/instantsearch.js/blob/feat/send-events-2/src/connectors/toggle-refinement/connectToggleRefinement.js#L141-L165)
- [connectRatingMenu](https://github.com/algolia/instantsearch.js/blob/feat/send-events-2/src/connectors/rating-menu/connectRatingMenu.js#L117:L140)

They all have slightly different implementation due to their own logic. So I just put them inside. I can extract them to util folder. It might improve the readability of the connectors, but won't be re-used somewhere else.

What do you think?

Reply:
I think for single-use ones, it might make sense to define this top-level in the connector's file, and once reused it can move to utils?

Reply:
done https://github.com/algolia/instantsearch.js/pull/4463/commits/ce4f5f0840dcbb893066e0ff40309b1f4175ea2b